packages/langium/src/test/langium-test.ts

```diff
@@ -9,16 +9,17 @@ import { TextDocument } from 'vscode-languageserver-textdocument';
 import { LangiumDocument } from '../documents/document';
 import { BuildResult } from '../documents/document-builder';
 import { LangiumServices } from '../services';
+import { AstNode } from '../syntax-tree';
 import { getDocument } from '../utils/ast-util';
 
-export function parseHelper(services: LangiumServices): (input: string) => BuildResult {
+export function parseHelper<T extends AstNode = AstNode>(services: LangiumServices): (input: string) => BuildResult<T> {
     const metaData = services.LanguageMetaData;
     const documentBuilder = services.documents.DocumentBuilder;
     return input => {
         const randomNumber = Math.floor(Math.random() * 10000000) + 1000000;
         const textDocument = TextDocument.create(`file:/${randomNumber}${metaData.fileExtensions[0]}`, metaData.languageId, 0, input);
-        const document: LangiumDocument = services.documents.LangiumDocumentFactory.fromTextDocument(textDocument);
-        const buildResult = documentBuilder.build(document);
+        const document = services.documents.LangiumDocumentFactory.fromTextDocument(textDocument) as LangiumDocument<T>;
+        const buildResult = documentBuilder.build<T>(document);
```

Comment:
Perhaps we should make `fromTextDocument` generic as well with `<T extends AstNode = AstNode>` so we don't need to cast here?